CMakeLists.txt

```diff
@@ -225,6 +225,8 @@ CHECK_INCLUDE_FILE("unistd.h"       HAVE_UNISTD_H)
 include(TestLargeFiles)
 OPJ_TEST_LARGE_FILES(OPJ_HAVE_LARGEFILES)
 
+set(CMAKE_REQUIRED_DEFINITIONS -D_POSIX_C_SOURCE=200112L)
```

Comment:
I'm not an cmake expert. How can I set that definition only for the posix_memalloc test?